src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -454,7 +455,10 @@ private uint ReportTcpSNIError(string errorMessage)
 
         private uint ReportErrorAndReleasePacket(SNIPacket packet, string errorMessage)
         {
-            packet.Release();
+            if (packet != null)
+            {
+                packet.Release();                
```

Comment:
I was going to comment here...but wasn't sure if the lib was using it at all yet, +1 if so.